packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -303,12 +285,14 @@ export function dispatchEventForPluginEventSystem(
       // TODO: useEvent for document and window
       return;
     }
-    // If we detect the FB legacy primer system, we
+    // If we are using the legacy FB support flag, we
     // defer the event to the null with a one
     // time event listener so we can defer the event.
     if (
-      enableLegacyFBPrimerSupport &&
-      willDeferLaterForFBLegacyPrimer(nativeEvent)
+      enableLegacyFBSupport &&
+      (eventSystemFlags & LEGACY_FB_SUPPORT) === 0 &&
```

Comment:
We want the opposite here, we want to not invoke the logic if the incoming event has been set with , otherwise we'll be endlessly making event listeners and get stuck in a cycle.

Reply:
Oh ok I get it. I think the structure of the comment confused me - "if we are using legacy, then we defer event", and I didn't notice  willDeferLaterForLegacyFBSupport is a boolean returning function, I though it was inside the block ha.